tests/udp.rs

```diff
@@ -559,3 +559,38 @@ fn localhost_ping_pong() -> Result {
     });
     sim.run()
 }
+
+#[test]
+fn socket_capacity() -> Result {
+    let mut sim = Builder::new()
+        .min_message_latency(Duration::from_millis(1))
+        .max_message_latency(Duration::from_millis(1))
+        .udp_capacity(1)
+        .build();
+
+    let (tx, rx) = oneshot::channel();
+
+    sim.client("server", async move {
+        let s = bind().await?;
+
+        _ = rx.await;
+        recv_ping(&s).await?;
+        assert!(timeout(Duration::from_secs(1), recv_ping(&s))
+            .await
+            .is_err());
+
+        Ok(())
+    });
+
+    sim.client("client", async move {
+        let s = bind().await?;
+
+        send_ping(&s).await?;
+        send_ping(&s).await?; // dropped
```

Comment:
how does this test that the packet was dropped? this is suppose to hit L182 of host.rs, correct?

Reply:
We send 2 datagrams and then notify the peer to recv them. With a capacity of 1, the first is delivered and enqueued and the second is dropped. Line 578 verifies that the second was never delivered by timing out on recv.

Reply:
oh gotcha. thanks!